src/docfx/serve/lsp/TextDocumentHandler.cs

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server.Capabilities;
+using OmniSharp.Extensions.LanguageServer.Protocol.Window;
+
+namespace Microsoft.Docs.Build
+{
+    internal class TextDocumentHandler : ITextDocumentSyncHandler
+    {
+        private readonly ILogger<TextDocumentHandler> _logger;
+        private readonly ILanguageServerConfiguration _configuration;
```

Comment:
nit: remove `_logger` and `_configuration` since they are not used in this PR. Inject them later when needed.